update webpack spit chunks configuration

scytacki opened the pull request:
This update slightly reduces the size of the main chunk. More importantly it makes the chunks more clear.

We could optimize further by splitting the `vendor-main` chunk up. However I think the best next step is to remove any dependency on slate and slate-editor from the main codebase. The slate-editor is our largest dependency currently, and I think it is only really needed by the text tile.

When looking at this new split it would be best look at what is inside of each module. I don't have a nice way to show that in this PR. The wepback-split-chunks.md file documents the two commands needed to look at it locally.

kswenson reviewed (changes requested):
👍 This looks great! The bundles are much more useful than the previous bundles that resulted from the example I had been following previously. I just have a couple of suggestions for tweaks to names of the chunks and the usual assortment of typographical/grammatical suggestions that you can do with as you see fit.

scytacki commented:
@kswenson I think I addressed all of your comments. And I spent several more hours figuring out how the cacheGroups work so I could create the vendor-main chunk without also breaking apart the tile chunks.  I think I now have a firm grasp on how it all works, so could customize it further if we want to tweak things.

kswenson reviewed (approved):
👍 Now it looks even greater!

scytacki commented:
I just realized I didn't address you comment about css. It seems like they are not getting split which seems OK to me. I'll add a note.

kswenson commented on docs/webpack-split-chunks.md:
```diff
@@ -0,0 +1,144 @@
+| 696B | SharedVariables.b0ce62b2.css |
+| 207B | Starter.fdf39778.css |
+
+The SplitChunksPlugin configuration is not doing any special to make this happen. Their filename format is coming from the `MiniCssExtractPlugin`.
```
```suggestion
The SplitChunksPlugin configuration is not doing anything special to make this happen. Their filename format is coming from the `MiniCssExtractPlugin`.
```

kswenson reviewed (approved):
👍 Thx for the additional CSS information
